File: Course_material/Java/DetectingEnglishText/src/com/globalsoftwaresupport/LanguageDetector.java
package com.globalsoftwaresupport;

import java.util.List;

public class LanguageDetector {
	
	private FileProcessor fileProcessor;
	// we store the English words in an array (ArrayList)
	private List<String> englishWords;
	
	public LanguageDetector() {
		this.fileProcessor = new FileProcessor();
		this.englishWords = fileProcessor.getWords();
	}

	// counts the number of English words in a given text
	private int countEnglishWordsIntext(String text) {
		//upper case letters are needed
		text = text.toUpperCase();
		
		// transform the text into a list of words (split by spaces)
		String[] words = text.split(" ");
		// matches counts the number of English words in the text
		int matches = 0;
		
		for(String s : words)
			if(englishWords.contains(s))
				matches++;
		
		return matches;
	}
	
	public boolean isEnglish(String text) {
		// number of English words in a given text
		int matches = countEnglishWordsIntext(text);
		
		// you can define your own classification algorithm
		// in this case the assumption: if 60% of the words in 
		// the text are English words then it is an English text
		if( (float) matches / text.split(" ").length * 100 >= 60)
			return true;
		
		return false;
	}
}

File: Caesar_Cipher/language_detector.py

ENGLISH_WORDS = []

def get_data():
    dictionary = open('./Caesar_Cipher/english_words.txt', 'r')
    for word in dictionary.read().split('\n'):
        ENGLISH_WORDS.append(word)
    dictionary.close()

def count_word(text):
    text = text.upper()
    words = text.split(' ')
    matches = 1
    for word in words:
        if word in ENGLISH_WORDS:
            matches += 1
    return matches

def is_english_text(text):
    matches = count_word(text)

    # If 80% of words are english, then we can say it's English
    print((float(matches)/len(text.split(' '))) * 100)
    if (float(matches)/len(text.split(' '))) * 100 > 80: 
        return True
    return False

if __name__ == '__main__':
    get_data()
    text = 'My Name is Shittu'
    print(is_english_text(text))